### PR TITLE
feat: add delete confirmation dialog before real delete to prevent unexpected operation

### PR DESCRIPTION
**Pull request by xinatcg**
image

currently, all modal messages are the same, but the service supports custom titles and content for different usage.

https://github.com/heroiclabs/nakama/issues/1123

### CODE REVIEW

File: console/ui/src/app/shared/delete-confirm.service.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  providedIn: 'root'
+})
+export class DeleteConfirmService {
+
+  constructor(private modalService: NgbModal) {
+  }
+
+  openDeleteConfirmModal(confirmedCallback: () => void, title: string = '', message: string = ''): void {
+    const modalOptions: NgbModalOptions = {
+      backdrop: false,
```
Comment (sesposito) [inline console/ui/src/app/shared/delete-confirm.service.ts:15]: I suggest to add:
```
centered: true
```

**Review by sesposito — CHANGES_REQUESTED** [review]
Thank you very much for your contribution! 
The PR looks good, however, if possible, could you also replace the usage of the `confirmDelete` modal in config and chatMessages with the same approach?
Before this can be merged, you'll also need to run `npm run-script build`

File: console/ui/src/app/shared/delete-confirm.service.ts (hunk 1 above)
Comment (xinatcg) [reply console/ui/src/app/shared/delete-confirm.service.ts:15]: ok, np thanks

**Comment by xinatcg** [comment]
> The PR looks good, however, if possible, could you also replace the usage of the `confirmDelete` modal in config and chatMessages with the same approach?

Ok, if so
> The PR looks good, however, if possible, could you also replace the usage of the confirmDelete modal in config and chatMessages with the same approach?

Should we leave them out and only keep the code for others without confirmation?

**Comment by sesposito** [comment]
I'll approve without those changes, but it would be good to touch on them as part of this PR, if possible. I think those modals aren't being displayed correctly due to the upgrade to Angular 15 merge.

**Comment by xinatcg** [comment]
> I'll approve without those changes, but it would be good to touch on them as part of this PR, if possible. I think those modals aren't being displayed correctly due to the upgrade to Angular 15 changes.

make some updates to support formGroup with 2 control in the confirm component

File: console/ui/src/app/config/config.component.ts [hunk 2]
```diff
@@ -153,29 +155,27 @@ export class ConfigComponent implements OnInit, OnDestroy {
...
+            this.deleteError = '';
+            this.deleteSuccess = true;
+          }, err => {
+            this.deleting = false;
+            this.deleteError = err;
+          },
+        );
       },
+      this.confirmDeleteForm,
+      'Delete Project (NOT Configuration)' ,
```
Comment (sesposito) [inline console/ui/src/app/config/config.component.ts:174]:
```suggestion
      'Delete All Data' ,
```

File: console/ui/src/app/config/config.component.ts [hunk 3]
```diff
@@ -153,29 +155,27 @@ export class ConfigComponent implements OnInit, OnDestroy {
...
+      'Are you sure you want to delete all data?'
```
Comment (sesposito) [inline console/ui/src/app/config/config.component.ts:175]:
```suggestion
     'Are you sure you want to delete all the database data?'
```
This should make it clearer, what do you think?

**Review by sesposito — APPROVED** [review]
Left a couple of suggestions but otherwise LGTM 👍

**Comment by xinatcg** [comment]
> Left a couple of suggestions but otherwise LGTM 👍

update with 2 suggestions. thanks

**Comment by sesposito** [comment]
Please rerun `npm run build` and we'll get this merged 👍

**Comment by xinatcg** [comment]
> npm run-script build

done

**Comment by xinatcg** [comment]
> Please rerun `npm run build` and we'll get this merged 👍

done

**Pull request merged**